docs/core/design/dcrypt.md

```diff
@@ -109,30 +115,206 @@ asymmetric key pair. File encryption can be done using whatever
+macl - eof  message integrity tag
 ```
 
+## Decryption Script
+
 There is a small script for decrypting these files, see
-[`dcrypt-decrypt.rb`](https://github.com/dovecot/tools/blob/main/dcrypt-decrypt.rb).
+[`dcrypt-decrypt.py`][dcrypt-decrypt].
+
+### Dependencies
+
+First fetch the script by cloning the [dovecot-tools][dovecot-tools]
+repository or fetching the single tool.
+
+::: info
+To clone the repository:
+
+```sh
+git clone https://github.com/dovecot/tools
+```
+
+To fetch only the script (e.g. by using `wget`):
+
+```sh
+wget https://raw.githubusercontent.com/dovecot/tools/refs/heads/main/dcrypt-decrypt.py
+```
+:::
+
+Setup a local python environment (optional) and install the necessary
+dependencies.
+
+```sh
+# Optional but recommended: Create a virtual environment for the necessary
+# python dependencies to run this script.
+python -m venv venv
+source venv/bin/activate
+# Install the dependencies.
+pip install asn1 cryptography
+```
+
+### Operation
+
+Probably the most common installation uses the [[plugin,mail-compress]] and
+[[plugin,mail-crypt]] plugins for compression and encryption respectively. See
+also [[link,fs_crypt]] for this setup.
+
+To simulate this scenario consider storing an input file `$in_file` using:
+- `$private_key_path` and `$public_key_path` - the private and public key files
+  to allow encryption with,
+- `$prefix` - the path prefix to use for storing the output file,
+- `$out_file` - the file path to store the file as.
+
+```sh
+doveadm fs put compress \
+    gz:6:crypt:private_key_path=$private_key_path:public_key_path=$public_key_path:posix:prefix=$prefix \
+    $in_file $out_file
+```
+
+Using the `dcrypt-decrypt` script you can then retrieve the key and file
+information using the following command. Make sure to supply the same
+`$private_key_path`, `$prefix` and `$out_file` as above:
+
+```sh
+dcrypt-decrypt.py -i -k $private_key_path -f ${prefix}${out_file} | gunzip
+```
+
+::: info
+The piped `gunzip` will ensure the contents are decompressed after decryption.
+:::
+
+### Output Description
+
+Run `dcrypt-decrypt.py --help` to read the tool's help text, which gives hints
+about its supported flags and its output.
+
+Supplying the private key file via the `-k`/`--key` flag will decrypt the
+encrypted file contents.
+
+::: info
+Fields that contain binary data are represented in hexadecimal form. Other
+fields are converted or displayed in human readable form.
+:::
+
+#### File information
+
+The `-i`/`--info` flag outputs a list of fields of the supplied file:
+
+`Version`
+:   What version of the dovecot [[link,lib_dcrypt_key_formats,Key Formats]] has
+    been used to encrypt this file.
+
+`Flags`
+:   What [[link,lib_dcrypt_flags,Flags]] have been used when encrypting the
+    file.
+
+`Header length`
+:   Number of bytes in the header.
+
+`Cipher algo`
+:   Name and Object Identifier (OID) of the cipher used for encrypting the
+    key - and data if it was used for both.
```

Comment:
this is kinda implying the wrong way.

SAME_ALGO means that we encrypt the encryption key material with same algorithm as data, not the other way round. If SAME_ALGO is not used, we always use aes256-ctr for keying material.